stdlib/asm/crypto/stark/random_coin.masm

```diff
@@ -73,103 +75,119 @@ end
 
-    ## Compute trace_length
-    ## Cycles: 20
+    # Compute `trace_length` = `2^log2(trace_length)` and assert it is a u32 value
     dup
     pow2
     u32split assertz
-    #=> [trace_length, log(trace_length), num_queries, log(blowup), grinding]
+    #=> [trace_length, log(trace_length), num_queries, log(blowup), grinding, ...] (20 cycles)
 
-    ## Save the trace length and its log to memory
+    # Save the `trace_length` to memory
     dup.0 exec.constants::trace_length_ptr mem_store
+    #=> [trace_length, log(trace_length), num_queries, log(blowup), grinding, ...] (3 cycles)
 
-    ## Assert blowup is equal to 8
-    ##  Cycles: 6
-    swap
+    # Assert blowup is equal to 8
     dup.3
-    dup
-    push.3
+    push.LOG_BLOWUP
     assert_eq
+    #=> [trace_length, log(trace_length), num_queries, log(blowup), grinding, ...] (3 cycles)
 
-    ## Compute log(lde_size) and lde_size and store them
-    ## Cycles: 32
-    add
+    # Compute `log(lde_size)` = `log(blowup) + log(trace_length)`
     swap
     movup.3
-    pow2
+    add
+    #=> [log(lde_size), trace_length, num_queries, grinding, ...] (3 cycles)
+
+    # Compute `blowup_size` = `2^log(blowup)`
+    swap
+    push.BLOWUP_SIZE
+    #=> [blowup_size, trace_length, log(lde_size), num_queries, grinding, ...] (2 cycles)
+
+    # Compute `lde_size` = `trace_length*blowup_size`
     dup
     movdn.4
     swap
     dup
     movdn.3
     mul
+    #=> [lde_size, trace_length, log(lde_size), blowup_size, num_queries, grinding, ...] (6 cycles)
+
+    # Save `[lde_size, log(lde_size), 0, 0]` to memory
     push.0
     movdn.2
     push.0
     movdn.2
     exec.constants::lde_size_ptr mem_storew
-    #=> [lde_size, log(lde_size), 0, 0, trace_length, num_queries, blowup, grinding]
-
+    #=> [lde_size, log(lde_size), 0, 0, trace_length, num_queries, blowup, grinding, ...] (6 cycles)
 
-    # Construct the proof context
-    # Cycles: 9
+    # clean top of the stack
     dropw
+    #=> [trace_length, num_queries, blowup, grinding, ...] (4 cycles)
 
-    ##trace layout info
-    push.1224804624
+    # Create the initial seed for randomness generation from proof context
+    # ---------------------------------------------------------------------------------------------
 
-    ##field modulus bytes (2 field elements)
+    # Construct the proof context
+    # trace layout info
+    push.1224804624
+    # field modulus bytes (2 field elements)
     push.1
     push.4294967295
-
-    ## field extension and FRI parameters
+    # field extension and FRI parameters
     push.132103
-
+    #=> [fri_parameters, M_1, M_0, trace_layout_info, trace_length, num_queries, blowup, grinding, ...] (4 cycles)
 
     # Hash proof context
-    # Cycles: 15
+    # A=`[trace_length, num_queries, blowup, grinding]`
+    # B=`[fri_parameters, M_1, M_0, trace_layout_info]`
+
+    # push capacity
     swapw
     push.1.0.0.0
-    #=> [0, 0, 0, 1, B, A, ..]
+    #=> [0, 0, 0, 1, B, A, ..] (5 cycles)
```

Comment:
Question: Why are we setting the padding bit in the capacity word?

Reply:
It is because the hashing is continued in `exec.public_inputs::load` and the total number of hashed elements is not divisible by 8 as can be seen here https://github.com/0xPolygonMiden/miden-vm/blob/fb648a4b6098e3574c4de4b53dac001ffc850d0e/stdlib/asm/crypto/stark/public_inputs.masm#L31

Reply:
Aha! Thanks for the explanation :)